src/kontrol/solc_to_k.py

```diff
@@ -411,6 +410,47 @@ def callvalue_cell(self) -> KInner:
                 else abstract_term_safely(KVariable('_###CALLVALUE###_'), base_name='CALLVALUE')
             )
 
+        @cached_property
+        def encoded_args(self) -> tuple[KInner, list[KInner]]:
+            args: list[KInner] = []
+            type_constraints: list[KInner] = []
+            for input in self.inputs:
+                abi_type = input.to_abi()
+                args.append(abi_type)
+                rps = []
+                if input.type.startswith('tuple'):
+                    components = input.components
+
+                    if input.type.endswith('[]'):
+                        if input.array_lengths is None:
+                            raise ValueError(f'Array length bounds missing for {input.name}')
+
+                        tuple_array_components = [
+                            Input(
+                                f'{_c.name}_{i}',
+                                _c.type,
+                                _c.components,
+                                _c.idx,
+                                _c.array_lengths,
+                                _c.dynamic_type_length,
+                            )
+                            for i in range(input.array_lengths[0])
+                            for _c in components
+                        ]
+                        components = tuple(tuple_array_components)
+
+                    for sub_input in components:
+                        _abi_type = sub_input.to_abi()
+                        rps.extend(_range_predicates(_abi_type, sub_input.dynamic_type_length))
+                else:
+                    rps = _range_predicates(abi_type, input.dynamic_type_length)
```

Comment:
Could this be a place to bring in the string constraints that we need? They'd need to be written using `#lookup`, so we need the storage and slot information...

Reply:
Yes, I think so! We should have access to the storage and slot information from [`Contract.fields`](https://github.com/runtimeverification/kontrol/blob/2fc32a5938beff42730dccc0548d30ba6969426d/src/kontrol/solc_to_k.py#L788), for example — we can pass `contract: Contract` parameter to this function; or just add this info to the `Input` object directly maybe (cc @nwatson22)